From b13da8a42a01d3d7e2a71bcae5d9aceecdcb6e82 Mon Sep 17 00:00:00 2001
From: Calum Lind <calumlind+deluge@gmail.com>
Date: Sun, 29 Oct 2017 10:46:26 +0000
Subject: [#3010|GTKUI] Handle unknown OverflowError from twisted reactor

---
 deluge/ui/gtkui/gtkui.py | 7 ++++++-
 1 file changed, 6 insertions(+), 1 deletion(-)

diff --git a/deluge/ui/gtkui/gtkui.py b/deluge/ui/gtkui/gtkui.py
index cd6fcc8..eecd026 100644
--- a/deluge/ui/gtkui/gtkui.py
+++ b/deluge/ui/gtkui/gtkui.py
@@ -291,7 +291,12 @@ class GtkUI(object):
         reactor.callWhenRunning(self._on_reactor_start)
         # Start the gtk main loop
         gtk.gdk.threads_enter()
-        reactor.run()
+        try:
+            reactor.run()
+        except OverflowError:
+            # Ticket 3010 reports an error that cannot replicate so catch
+            # it and ignore it to prevent spamming logs.
+            pass
         self.shutdown()
         gtk.gdk.threads_leave()
 
-- 
cgit v1.1

From 396417bcd045c387fd5ee6cae6a18106324a06a4 Mon Sep 17 00:00:00 2001
From: Calum Lind <calumlind+deluge@gmail.com>
Date: Sun, 29 Oct 2017 11:12:56 +0000
Subject: [#3124|GTKUI] Fix comparing Name str if value is None

The original fix was not correct as the strcoll function cannot
accept None only strings. This fix ensures that the value is an
empty string if None for comparison.
---
 deluge/ui/gtkui/gtkui.py       |  4 ++--
 deluge/ui/gtkui/torrentview.py | 19 +++++++++----------
 2 files changed, 11 insertions(+), 12 deletions(-)

diff --git a/deluge/ui/gtkui/gtkui.py b/deluge/ui/gtkui/gtkui.py
index eecd026..14954b1 100644
--- a/deluge/ui/gtkui/gtkui.py
+++ b/deluge/ui/gtkui/gtkui.py
@@ -294,8 +294,8 @@ class GtkUI(object):
         try:
             reactor.run()
         except OverflowError:
-            # Ticket 3010 reports an error that cannot replicate so catch
-            # it and ignore it to prevent spamming logs.
+            # Ticket #3010 reports an overflow error from twisted glibbase that
+            # cannot be replicated so catch it and ignore it to prevent spamming logs.
             pass
         self.shutdown()
         gtk.gdk.threads_leave()
diff --git a/deluge/ui/gtkui/torrentview.py b/deluge/ui/gtkui/torrentview.py
index 6befddb..b210153 100644
--- a/deluge/ui/gtkui/torrentview.py
+++ b/deluge/ui/gtkui/torrentview.py
@@ -171,18 +171,17 @@ def cell_data_queue(column, cell, model, row, data):
         cell.set_property("text", str(value + 1))
 
 def str_nocase_sort(model, iter1, iter2, data):
-    """
-    Sort string column data with locale.strcoll which (allegedly)
-    uses ISO 14651.
+    """Sort string column data using ISO 14651 in lowercase.
+
+    Uses locale.strcoll which (allegedly) uses ISO 14651. Compares first
+    value with second and returns -1, 0, 1 for where it should be placed.
 
     """
-    try:
-        v1 = model[iter1][data].lower()
-        v2 = model[iter2][data].lower()
-    except AttributeError:
-        # Catch None type for value.
-        v1 = model[iter1][data]
-        v2 = model[iter2][data]
+    v1 = model[iter1][data]
+    v2 = model[iter2][data]
+    # Catch any values of None from model.
+    v1 = v1.lower() if v1 else ""
+    v2 = v2.lower() if v2 else ""
     return strcoll(v1, v2)
 
 def queue_peer_seed_sort_function(v1, v2):
-- 
cgit v1.1

