diff -up ./deluge/data/share/applications/deluge.desktop.old ./deluge/data/share/applications/deluge.desktop
--- ./deluge/data/share/applications/deluge.desktop.old	2010-10-13 19:54:05.713911783 -0700
+++ ./deluge/data/share/applications/deluge.desktop	2010-10-13 19:54:11.810912505 -0700
@@ -7,6 +7,6 @@ Exec=deluge-gtk
 Icon=deluge
 Terminal=false
 Type=Application
-Categories=Network;
+Categories=Network;P2P;
 StartupNotify=true
 MimeType=application/x-bittorrent;
